test/node/replumb/require_node_test.cljs

```diff
@@ -468,21 +479,47 @@ trim-newline
       (reset-env! '[my.namespace foo.bar.baz]))
 
     ;; (let [res (do (read-eval-call "(ns my.namespace (:require [foo.bar.baz :refer-macros [mul-baz]]))")
     ;;               (read-eval-call "(mul-baz 10 12)"))
     ;;       out (unwrap-result res)]
-    ;;   (is (success? res) "(ns my.namespace (:require ...)) and (mul 10 12) should succeed")
-    ;;   (is (valid-eval-result? out) "(ns my.namespace (:require ...)) and (mul 10 12) should be a valid result")
-    ;;   (is (= "120" out) "(mul 10 12) should be equal to 120")
+    ;;   (is (success? res) "(ns my.namespace (:require ...:refer-macros...)) and (mul-baz 10 12) should succeed")
+    ;;   (is (valid-eval-result? out) "(ns my.namespace (:require ...:refer-macros)) and (mul-baz 10 12) should be a valid result")
+    ;;   (is (= "120" out) "(mul-baz 10 12) should be equal to 120")
     ;;   (reset-env! '[my.namespace foo.bar.baz]))
-    )
 
+    (let [res (do (read-eval-call "(ns my.namespace (:require [foo.bar.core :refer-macros [mul-core]]))")
+                  (read-eval-call "(mul-core 10 20)"))
+          error (unwrap-result res)]
+      (is (not (success? res)) "(ns my.namespace (:require ...:refer-macros...)) and (mul-core 10 20) should not succeed")
+      (is (valid-eval-error? error) "(ns my.namespace (:require ...:refer-macros...)) and (mul-core 10 20) should be an instance of js/Error")
+      (is (re-find #"ERROR" (extract-message error))
+          "(ns my.namespace (:require ...:refer-macros...)) and (mul-core 10 20) should have correct error message")
+      (reset-env! '[my.namespace foo.bar.core foo.bar.macros]))
+
+    ;; (let [res (do (read-eval-call "(ns my.namespace (:require [foo.bar.core :refer-macros [mul-core]]))")
+    ;;               (read-eval-call "(mul-core 10 20)"))
+    ;;       out (unwrap-result res)]
+    ;;   (is (success? res) "(ns my.namespace (:require ...:refer-macros...)) and (mul-core 10 20) should succeed")
+    ;;   (is (valid-eval-result? out) "(ns my.namespace (:require ...:refer-macros...)) and (mul-core 10 20) should have a valid result")
+    ;;   (is (= "200" out ) "(mul-core 10 20) should produce 200")
+    ;;   (reset-env! '[my.namespace foo.bar.core foo.bar.macros]))
+)
+
+  (deftest ns-macro-self-requiring-namespace
+    ;; see "loop" section here: http://blog.fikesfarm.com/posts/2015-12-18-clojurescript-macro-tower-and-loop.html
+    ;; but it does not work in JS ClojureScript
+    (let [res (read-eval-call "(ns my.namespace (:require [foo.bar.self]))")
+          error (unwrap-result res)]
+      (is (not (success? res)) "(ns my.namespace (:require [foo.bar.self])) should not succeed")
+      (is (valid-eval-error? error) "(ns my.namespace (:require [foo.bar.self])) should be an instance of js/Error")
+      (is (re-find #"Maximum call stack size exceeded" (extract-message error)) "(ns my.namespace (:require [foo.bar.self])) should have correct error message")
```

Comment:
This is what I got back:
"I would guess if you try the loop example in bootstrapped ClojureScript it would recursively load `foo.core` with nothing to stop the recursion."
So yes, I think the error message is correct in this case, don't know in the future... but if something changes we'll see this test fail.